src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1897,6 +1949,118 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
+            using (GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+            {
+                var collectionName = GetCollectionFor(context, lowerIdPrefix);
+                if (collectionName == null)
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                    return 0;
+                }
+
+
+                if (ShouldAdoptRevision(context, lowerId, lowerIdPrefix, collectionName, out var table, out var lastRevision))
+                {
+                    var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+                    var newEtag = _database.DocumentsStorage.GenerateNextEtag();
+                    var cv = _documentsStorage.GetNewChangeVector(context, newEtag);
+                    return CreateDeleteRevision(context, table, id, collectionName, cv, lastModifiedTicks, lastRevision.Flags) - 1;
+                }
+
+                return 0;
+            }
+        }
+
+        internal class TestingStuff
+        {
+            private RevisionsStorage _parent;
+
+            public TestingStuff(RevisionsStorage revisionsStorage)
+            {
+                _parent = revisionsStorage;
+            }
+
+            internal void DeleteLastRevisionFor(DocumentsOperationContext context, string id, string collection)
+            {
+                var collectionName = new CollectionName(collection);
+                using (DocumentIdWorker.GetSliceFromId(context, id, out var lowerId))
+                using (_parent.GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+                using (GetKeyWithEtag(context, lowerId, etag: long.MaxValue, out var compoundPrefix))
+                {
+                    var table = _parent.EnsureRevisionTableCreated(context.Transaction.InnerTransaction, collectionName);
+                    var holder = table.SeekOneBackwardFrom(RevisionsStorage.RevisionsSchema.Indexes[RevisionsStorage.IdAndEtagSlice], lowerIdPrefix, compoundPrefix);
+                    var lastRevision = TableValueToRevision(context, ref holder.Reader, DocumentFields.ChangeVector | DocumentFields.LowerId);
+                    _parent.DeleteRevisionFromTable(context, table, new Dictionary<string, Table>(), lastRevision, collectionName, lastRevision.ChangeVector, _parent._database.Time.GetUtcNow().Ticks);
+                    IncrementCountOfRevisions(context, lowerIdPrefix, -1);
+                }
+            }
+        }
+
+        internal TestingStuff ForTestingPurposes;
+
+        internal TestingStuff ForTestingPurposesOnly()
+        {
+            if (ForTestingPurposes != null)
+                return ForTestingPurposes;
+
+            return ForTestingPurposes = new TestingStuff(this);
+        }
+
+        private unsafe long CreateDeleteRevision(DocumentsOperationContext context, Table table, string id, CollectionName collectionName, 
```

Comment:
CreateDeleteRevision -> CreateDeletedRevision

Reply:
https://github.com/ravendb/ravendb/pull/17355#discussion_r1342095990

Reply:
here you not deleting a revision, but you create one (a revision that represents deletion), so it should be `CreateDeletedRevision`.